### ISSUE TITLE
waitFor stores

### ISSUE THREAD

**Issue by ghost** [meteor-space/flux#1]
This flux implementation lacks in a way of waiting for other stores like shown in this example: https://facebook.github.io/flux/docs/dispatcher.html#example. The example also illustrates the use case. It's needed when a store depends on state of another store.

I have merged my stores for now to fix this problem in our app.

**Comment by DominikGuzei** [comment]
With reactivity in Meteor we usually don't need to wait for other stores. Could you post an example where you still need that?

**Comment by ghost** [comment]
Yeah, I guess I could just have depended on the state of the other store in an autorun block instead of trying to do everything in the event handler.

Here is my scenario that I had (when I remember correctly):

Two stores:
- Workspace Store: Stores the currently open file.
- Comments Store: Stores the comments for the currently open file.

There is an action/event "OpenFile" that is emitted by the view when the user clicks on a file to open it.

Both stores have a handler for "OpenFile".
- Workspace Store: Updates the current open file.
- Comments Store: Updates the comments for the new open file. Here I wanted to wait until the new file actually has been opened. So I would have done a waitFor on the Workspace Store event handler.

**Comment by DominikGuzei** [comment]
So the way we are doing this is by setting up `computations` which are setup as `Tracker.autorun` blocks automatically and thus are reactive. You can see an example of one store depending on another here:
https://github.com/meteor-space/donations/blob/develop/packages/app/source/client/stores/location-details-store.js#L24

To apply it for your example:

``` javascript
Space.flux.Store.extend('CommentsStore', {

  dependencies: {
    workspaceStore: 'WorkspaceStore'
  },

  reactiveVars() {
    return [{
      comments: []
    }];
  },

  computations() {
    return [this._updateComments];
  },

  _updateComments() {
    let openFile = this.workspaceStore.openFile();
    // Figure out smart stuff based on the open file
    this._setReactiveVar('comments', someValue);
  },
});
```

So you only have to handle the `OpenFile` event once in the workspace store :wink:
